solr/core/src/java/org/apache/solr/request/json/RequestUtil.java

```diff
@@ -68,8 +69,12 @@ public static void processParams(SolrRequestHandler handler, SolrQueryRequest re
       String[] jsonFromParams = map.remove(JSON);  // params from the query string should come after (and hence override) JSON content streams
 
       for (ContentStream cs : req.getContentStreams()) {
+        // if BinaryResponseParser.BINARY_CONTENT_TYPE, let the following fail below - we may have adjusted the content without updating the content type
+        // problem in this case happens in a few tests, one seems to happen with kerberos and remote node query (HttpSolrCall's request proxy)
+
         String contentType = cs.getContentType();
-        if (contentType==null || !contentType.contains("/json")) {
+        if (contentType==null || (!contentType.contains("/json") && !contentType.contains(
+            BinaryResponseParser.BINARY_CONTENT_TYPE))) {
```

Comment:
I saw that you switched a bunch of places away from BINARY_CONTENT_TYPE to multi-part -- does Jetty 10 not like octet-stream? What's the motivation?

Reply:
3 or 4 tests fail due to the search handler and it's content header check.